### PR TITLE
Feat: strip path prefixes

### PR DESCRIPTION
**Pull request by UnePierre**
## Description

This implements a good part of #882:

- Add the option `-spp` / `--strip-file-prefixes`.
- Document the option itself.
- Add and document a configurable DOCTEST_CONFIG_OPTIONS_FILE_PREFIX_SEPARATOR.
- Apply the meaningful path prefixes to the examples.

### CODE REVIEW

File: doc/markdown/commandline.md [hunk 1]
```diff
@@ -58,6 +58,7 @@ All the options can also be set with code (defaults/overrides) if the user [**su
 | ```-ns``` &nbsp; ```--no-skip=<bool>``` | Don't skip test cases marked as skip with a decorator |
 | ```-gfl``` ```--gnu-file-line=<bool>``` | ```:n:``` vs ```(n):``` for line numbers in output (gnu mode is usually for linux tools/IDEs and is with the ```:``` separator) |
 | ```-npf``` ```--no-path-filenames=<bool>``` | Paths are removed from the output when a filename is printed - useful if you want the same output from the testing framework on different environments |
+| ```-spp``` ```--skip-path-prefixes=<bool>``` | Remove the longest matching one of these prefixes from any file path in the output - useful as ```-npf```, but can preserve some context through relative paths. Try: ```--spp=${CMAKE_SOURCE_DIR}/:${CMAKE_BINARY_DIR}/``` |
```
Comment (onqtam) [inline doc/markdown/commandline.md:61]: this should not be a <bool> but a <string>

And perhaps should be changed from ```useful as -npf``` to ```similar to -npf``` and from ```but can preserve some context through relative paths``` to ```but can preserve some context by not removing the entire relative paths``` or something like that

Comment (UnePierre) [reply doc/markdown/commandline.md]: I'll also add a hint at the `:` separator.

**Comment by UnePierre** [comment]
Sorry, I wasn't quick enough in re-basing this pull request in order to add #881's new test's result.

**Comment by onqtam** [comment]
oh, sorry, I was too quick with the merge. yeah, open a new PR and I'll merge that one too.

**Comment by UnePierre** [comment]
> oh, sorry, I was too quick with the merge. yeah, open a new PR and I'll merge that one too.

My pleasure.

**Pull request merged**